readme.md

```diff
@@ -10,39 +10,130 @@ npm install gatsby-paginate --save
+Just insert a call to `createPaginatedPages` before (or after) the createPage function:
 
 ```javascript
-createPaginatedPages({ 
-  edges:  // An array of posts/edges, 
-  createPage, // The gatsby createPage function from boundActionCreators
-  pageTemplate, // The template file your pages will use 
-  pageLength // Optional number of posts to group/paginate by - default is 10
-  })
+exports.createPages = ({ graphql, boundActionCreators }) => {
+  const { createPage } = boundActionCreators;
+  return new Promise((resolve, reject) => {
+    graphql(`
+      //graphql query
+    `).then(result => {
+      createPaginatedPages({
+        edges: result.data.posts.edges,
+        createPage: createPage,
+        pageTemplate: "src/templates/index.js",
+        pageLength: 10
+      });
+      result.data.posts.edges.map(({ node }) => {
+        createPage({
+          path: node.fields.slug,
+          component: path.resolve("./src/templates/post.js"),
+          context: {
+            slug: node.fields.slug
+          }
+        });
+      });
+      resolve();
+    });
+  });
+};
 ```
 
-## Context ##
+Notice that `createPaginatedPages` is being passed an object. `edges` is the array of nodes that comes from the GraphQL query. `createPage` is simply the createPage function you get from `boundActionCreators`. `pageTemplate` is a template to use for the index page. And `pageLength` is an optional parameter that defines how many posts to show per index page. It defaults to 10. 
 
-This implementation negates the need for a separate `index.js` page. Instead it creates a context of `first` as a bool which returns `true` if the pagination index is 0. This can be used to conditionally render content on your first/index page inside the template specified as `pageTemplate`.
+`createPaginatedPages` will then call `createPage` to create an index page for each of the groups of pages. The content that describes the blogs (title, slug, etc) that will go in each page will be passed to the template through `props.pathContext` so you need to make sure that everything that you want on the index page regarding the blogs should be requested in the GraphQL query in `gatsby-node.js`.
 
-It also passes the index of current pagination, and a boolean value for last page. 
+### Create the index template
 
-A basic example of using context for conditionally rendering content on your index page would be like so...
+After deleting `index.js` from the pages directory, create an `index.js` file in templates. Add something like the following code to your template:
 
 ```javascript
-const { group, index, first, last } = pathContext;
-return (
-  <div>
-    {first ? (
-      <h1>This is an h1 for my homepage</h1>
-    ) : (
-      <h1>This is an h1 for subsequent pages</h1>
-    )}
-  </div>     
-```
\ No newline at end of file
+import React, {Component} from "react";
```

Comment:
> It may not be the prettiest code

Here's the [prettier](https://github.com/prettier/prettier) version of this snippet :) 
```javascript
import React, { Component } from "react";
import Link from "gatsby-link";
class IndexPage extends Component {
  render() {
    const data = this.props.data;
    const { group, index, first, last } = this.props.pathContext;
    return (
      <div>
        <h4>{data.allMarkdownRemark.totalCount} Posts</h4>
        {group.map(({ node }) => (
          <div key={node.id}>
            <h3>
              {node.frontmatter.title}
              <br />
            </h3>
          </div>
        ))}
        {!first ? (
          <Link to={index - 1 == 1 ? "" : (index - 1).toString()}>
            Go to Previous Page
          </Link>
        ) : (
          <span>Go to Previous Page</span>
        )}
        {!last ? (
          <Link to={(index + 1).toString()}>Go to Next Page</Link>
        ) : (
          <span>Go to Next Page</span>
        )}
      </div>
    );
  }
}
export default IndexPage;

export const pageQuery = graphql`
  query IndexQuery {
    allMarkdownRemark {
      totalCount
    }
  }
`;
```